src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.health -= 50;
+    }
```

Comment:
[CODE PERFORMANCE] After decreasing the health of the herbivore, you should check if its health is less than or equal to 0 and remove it from the Animal.alive array if necessary. This can be done using Array's `filter` method to avoid nesting loops which is not a performant solution.